snuba/subscriptions/worker.py

```diff
@@ -109,23 +117,80 @@ def __execute(
 
     def __execute_query(self, request: Request, timer: Timer) -> Tuple[Request, Result]:
         with self.__concurrent_gauge:
-            # XXX: The ``extra`` is discarded from ``QueryResult`` since it is
-            # not particularly useful in this context and duplicates data that
-            # is already being published to the query log.
-            # XXX: The ``request`` instance is copied when passed to
-            # ``parse_and_run_query`` since it can/will be mutated during
-            # processing.
-            return (
-                request,
-                parse_and_run_query(
+            is_consistent_query = request.settings.get_consistent()
+
+            def run_consistent() -> Result:
+                request_copy = copy.deepcopy(request)
+                request_copy.settings.set_consistent(True)
+
+                return parse_and_run_query(
+                    self.__dataset,
+                    request_copy,
+                    timer,
+                    robust=True,
+                    concurrent_queries_gauge=self.__concurrent_clickhouse_gauge,
+                ).result
+
+            def run_non_consistent() -> Result:
+                request_copy = copy.deepcopy(request)
+                request_copy.settings.set_consistent(False)
+
+                return parse_and_run_query(
                     self.__dataset,
                     copy.deepcopy(request),
                     timer,
                     robust=True,
                     concurrent_queries_gauge=self.__concurrent_clickhouse_gauge,
-                ).result,
+                ).result
+
+            def selector_func(run_snuplicator: bool) -> Tuple[str, List[str]]:
+                primary = "consistent" if is_consistent_query else "non_consistent"
+                other = "non_consistent" if is_consistent_query else "consistent"
+
+                return (primary, [other] if run_snuplicator else [])
+
+            def callback_func(
+                data: List[ThreadedFunctionDelegatorResult[Result]],
+            ) -> None:
+                primary_result = data.pop(0).result
+
+                for result in data:
+                    match = result.result == primary_result
+                    self.__metrics.increment(
+                        "consistent",
+                        tags={
+                            "dataset": self.__dataset_name,
+                            "consistent": str(is_consistent_query),
```

Comment:
Not sure what this tag is meant for.

Reply:
This is supposed to mark whether the primary query is a consistent one or not, since it's supposed to work independently of `consistent_override`. I wasn't sure if that was going to be removed or not so this makes no assumptions about that.

Reply:
Ok, got it. 
As I mention in another comment, I would not necessarily make this work independently: if we reduce the consistency via `consistent_override` we do not want to increase it with this. 
But you do not need to reference that flag explicitly in this code. You can just run a parallel query only if the original one is consistent to begin with.